lemmatization/services/chinese.py

```diff
@@ -69,19 +69,27 @@ def tokenize(self, text: str) -> TripleIterable:
     def normalize_chinese(self, text: str) -> str:
         """
         Performs Unicode normalization on text. cf Unicode Standard section 2.12.
-        Uses NFC (Normalization Form C). Also selectively converts full-width
-        pipe (｜ \uFF5C) and underscore (＿ \uFF3F) to half-width (ASCII) | \u007C
-        and _ \u005F (so they can be used to give token boundary hints).
-
-        >>> normalize_chinese("1｜2＿3")
-        "1|2_3"
-
-        There are some issues with using NFKC, e.g.:
+        Uses NFC (Normalization Form C).
+
+        Also:
+        - selectively converts full-width pipe (｜ \uFF5C) and underscore (＿ \uFF3F)
+        to half-width (ASCII) | \u007C and _ \u005F (so they can be used to give token
+        boundary hints).
+        - converts non-breaking spaces (which are sometimes present after editing an
+        existing text, due to the UI component employed) into regular spaces (\u00a0
+        to \u0020); the typesetting information is lost, but we don't need it for our
+        purposes, and it can complicate tokenization.
```

Comment:
Note that this only changes the way the Chinese tokenizer/lemmatizer deals with non-breaking spaces; I didn't test how they affect the other tokenizers, but I expect it's different due to spaces being used as delimiters for tokenization anyway.

Reply:
I also noted that pipes and spaces work in editing mode to split tokens, but underscores cannot be used to combine them. For example, try editing an existing text and splitting a multi-character word using a space or pipe and saving; they will now be separate lemmas in the text. However, if you then re-edit and add an underscore and/or remove the mediating space, it will not be merged as expected into a single lemma.